### PR TITLE
GET /api/reviews/:review_id

### PR DESCRIPTION
**Pull request by veryswagilikeit**

### CODE REVIEW

File: models/model.js [hunk 1]
```diff
@@ -4,4 +4,22 @@ exports.selectCategories = () => {
...
+};
+
+exports.selectReviewById = (id) => {
+    return db
+        .query(`SELECT * FROM reviews WHERE reviews.review_id = $1;`, [id])
+        .then(({ rows }) => {
+            return rows[0];
+        })
+        .then((review) => {
+            if (review === undefined) {
```
Comment (dappernerddesigns) [inline models/model.js]: This is very close, I'd be looking at conditional logic for the row count rather than checking if something is undefined.

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
This is very close on the error handling portion in your model. I'd have a look at what happens if rows.length is zero. You are also missing an update to your endpoints json file. Make these two amendments and please resubmit.

File: models/model.js [hunk 2]
```diff
@@ -4,4 +4,22 @@ exports.selectCategories = () => {
...
+        .then((review) => {
+            if (!review.length) {
+                return Promise.reject({
+                    status: 404,
+                    msg: "Not Found",
+                });
+            } else {
+                return review[0];
+            };
+        });
```
Comment (Slyose) [inline models/model.js:24]: Nicely done!

**Review by Slyose — APPROVED** [review]
Feedback all taken on board, happy to see this merged

**Pull request merged**